rs/execution_environment/src/scheduler/round_schedule.rs

```diff
@@ -408,7 +412,7 @@ impl RoundSchedule {
 
         // Compute the priority of the canisters for this round.
         let mut accumulated_priority_invariant = AccumulatedPriority::default();
-        let mut accumulated_priority_deviation = 0;
+        let mut accumulated_priority_deviation: i128 = 0;
```

Comment:
Since all metrics end up as `f64`; and there's absolutely no need for this to be exceedingly precise; I'd just use `f64`. Then the code would be a lot simpler and there would be no need for saturating this or saturating that.